beamer/config/commands.py

```diff
@@ -131,3 +138,255 @@ def read(
+    required=True,
+    help="Path to the directory with contract ABIs.",
+)
+@click.option(
+    "--artifact",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    help="Path to the deployment artifact.",
+)
+@click.option(
+    "--keystore-file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    metavar="PATH",
+    help="Path to the keystore file.",
+)
+@click.password_option(
+    "--password",
+    type=str,
+    default="",
+    prompt=False,
+    help="The password needed to unlock the keystore file.",
+)
+@click.argument(
+    "current_state_path",
+    type=click.Path(file_okay=True, dir_okay=False, exists=True, path_type=Path),
+)
+@click.argument(
+    "desired_state_path",
+    type=click.Path(file_okay=True, dir_okay=False, exists=True, path_type=Path),
+)
+def write(
+    rpc_file: Path,
+    abi_dir: Path,
+    artifact: Path,
+    keystore_file: Path,
+    password: str,
+    current_state_path: Path,
+    desired_state_path: Path,
+) -> None:  # pylint: disable=unused-argument
+    """Read contract configuration from CURRENT_STATE_PATH and DESIRED_STATE_PATH
+    and issue transactions so that the on-chain configuration matches the desired one."""
+    beamer.util.setup_logging(log_level="DEBUG", log_json=False)
+
+    account = beamer.util.account_from_keyfile(keystore_file, password)
+    log.info("Loaded keystore file", address=account.address)
+
+    rpc_info = beamer.deploy.config.load_rpc_info(rpc_file)
+    deployment = Deployment.from_file(artifact)
+
+    assert deployment.chain is not None
+    chain_id = deployment.chain.chain_id
+    url = rpc_info[chain_id]
+    w3 = make_web3(url, account)
+    assert w3.eth.chain_id == chain_id
+    log.info("Connected to RPC", url=url)
+
+    current_config = Configuration.from_file(current_state_path)
+    desired_config = Configuration.from_file(desired_state_path, False)
+
+    _ensure_config_block_not_in_future(w3, current_config)
+    _ensure_same_tokens_have_same_addresses(current_config, desired_config)
+    _ensure_same_chain_ids(current_config, desired_config)
+
+    # Current configuration is only valid for current_config.block.
+    # This means we need to make sure that there were no config updates
+    # in block range [current_config.block + 1, latest_block].
+    start_block = BlockNumber(current_config.block + 1)
+    _ensure_no_config_updates_since(w3, deployment, start_block)
+
+    receipt = None
+    for contract, function, *args in _generate_updates(current_config, desired_config):
+        log.info("Sending transaction", call=f"{contract}.{function}({', '.join(map(str, args))})")
+        contract = deployment.obtain_contract(w3, "chain", contract)
+        call = getattr(contract.functions, function)(*args)
+        try:
+            receipt = beamer.util.transact(call)
```

Comment:
nitpick and not really related: I was confused what transact can actually return. 
Return type is Any but looking at the function, it can only return `TxReceipt`. Wouldn't it be a good time to change that?